// my-component-debug/dist/my-component.js
import { defineCustomElement as __stencil_defineCustomElement, HTMLElement } from "@stencil/core/internal/client";
import { h } from "@stencil/core/internal/client";
import MyComponentStyle0 from "./my-component.css?tag=my-component-debug&encapsulation=shadow";
function format(first, middle, last) {
    return (first || '') + (middle ? ` ${middle}` : '') + (last ? ` ${last}` : '');
}
export const MyComponent = class extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.first = undefined;
        this.middle = undefined;
        this.last = undefined;
    }
    getText() {
        return format(this.first, this.middle, this.last);
    }
    render() {
        return h("div", { key: '3baac4815aa7da07fd44cb293022cdcc2a0e0f09', class: "testCSS" },
            "Hello, World! I'm ",
            this.getText());
    }
    static get style() { return MyComponentStyle0; }
};
__stencil_defineCustomElement(MyComponent, [1, "my-component-debug", {
        "first": [1],
        "middle": [1],
        "last": [1]
    }]);
